asdf/_tests/test_generic_io.py

```diff
@@ -394,7 +393,7 @@ def test_invalid_obj(tmp_path):
 
 
 def test_nonseekable_file(tmp_path):
-    base = io.IOBase
+    base = io.FileIO
```

Comment:
I'm not sure why `IOBase` wasn't failing with 3.12.rc1 but it fails with 3.12.dev. I don't see any description of this change in the release notes or PRs. My suspicion is the use of `IOBase` here was working in part because of a bug that is fixed in 3.12 (as providing a filename to `IOBase.__init__` doesn't really make sense so I could see how this could be fixed as bug during related `io` cleanup).